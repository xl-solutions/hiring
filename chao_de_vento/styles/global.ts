import { createGlobalStyle } from 'styled-components'

export default createGlobalStyle`
  :root {
    box-sizing: border-box;
  }

  *,
  ::before,
  ::after {
    box-sizing: inherit;
    margin: 0;
    padding: 0;
  }

  body {
    padding: 0;
    margin: 0;
    font-family: 'KoHo', sans-serif;
    line-height: 1.6;
    font-size: 1rem;
  }

  h1, h2, h3 {
    font-family: 'Girassol', cursive;
    text-align: center;
  }

  a {
  color: #0070f3;
  text-decoration: none;
  }

  a:hover {
    text-decoration: underline;
  }

  img {
    max-width: 100%;
    display: block;
  }
`
