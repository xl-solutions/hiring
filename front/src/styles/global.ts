import { createGlobalStyle } from 'styled-components';

import GithubLogo from '../assets/GithubLogo.svg';

export default createGlobalStyle`
*{
  margin:0;
  padding:0;
  box-sizing: border-box;
  outline: 0;
  list-style-type: none;
  background-size: cover;
}
body{
  background: #272424 url(${GithubLogo}) no-repeat 70% top;
  -webkit-font-smoothing: antialiased;
}
body, input, button{
  font-family: 'Roboto', sans-serif;
}
h1,h2,h3,h4,h5,h6,strong{
  font-weight: 500;
}
button {
    cursor: pointer;
  }

#root{
        margin: 0 auto;
        padding: 40px 20px;
}
`;
