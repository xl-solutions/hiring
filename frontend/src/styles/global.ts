import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
    :root {

        --blue: #0d6efd;
        --grey: #a0a6af;

        --text-title: #363f5f;
        --text-body: #969cb3;

        --background: #e7e7e7;
        --shape: #FFFFFF;
    }
    *{
        margin: 0;
        padding: 0;
        box-sizing: border-box;
    }

    html{
        @media (max-width: 1080px){
            font-size: 93.75%;
        }
         @media(max-width: 720px){
             font-size: 87.5%;
         }
    }

    body {
        background: var(--background);
        /* -webkit-font-smoothing: antialiased */
    }

    body, input, button {
        font-family: sans-serif;
        font-weight: 400;
    }

    h1, h2, h3, h4{
        font-weight: 600;
    }

    button {
        cursor: pointer;
    }

    [disabled] {
        opacity: 0.6;
        cursor: not-allowed;
    }


    .react-modal-overlay{
        background: rgba(0,0,0,0.5);

        position: fixed;
        top:0;
        bottom: 0;
        right: 0;
        left: 0;

        display: flex;
        align-items: center;
        justify-content: center;
    }
    .react-modal-content{
        width: 100%;
        height: 25rem;
        max-width: 70rem;
        background: var(--grey);
        padding: 3rem;
        position: relative;
        border-radius: 0.5rem;


        padding: 1rem 1rem;
        text-align: center;
        
    }
    .react-modal-close{
        position: absolute;
        right: 1.5rem;
        top: 1.5rem;
        border: 0;
        cursor: pointer;
    }

`
