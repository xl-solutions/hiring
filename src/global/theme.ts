const uiColors = {
  'success-default': '#2BAF1F',
  'success-lighter': '#F2FCF0',

  'info-default': '#0497F9',
  'info-lighter': '#F2FAFC',

  'warning-default': '#FFBF00',
  'warning-lighter': '#FFFAF2',

  'danger-default': '#F53636',
  'danger-lighter': '#FFF5F7',

  'white-default': 'rgba(255, 255, 255, 1)',
  'back-default': 'rgba(0, 0, 0, 1)',
};

const neutralColors = {
  dark: {
    'dark-darker': '#1E1E26',
    'dark-dark': '#2C2C36',
    'dark-default': '#3D3D4A',

    'dark-light': '#565666',
    'dark-lighter': 'rgba(114, 114, 133, 1)',
    'dark-disable': 'rgba(230, 230, 235, 1)',
  },
  light: {
    'light-darker': '#B2B2BD',
    'light-disabled': '#CECED6',
    'light-default': '#E6E6EB',

    'light-light': '#F0F0F5',
    'light-lighter': '#FAFAFC',
  },
};

const gridScale = {
  xxxs: 8,
  xxs: 12,
  xs: 16,
  md: 20,
  lg: 24,
  xl: 28,
  xxl: 32,
  xxxl: 36,
  full: 100,
  display: 40,
  '1/3': 33.33,
};

const componentScale = {
  total: 100,
  '1/4': 33.33,

  'card-width-default': 100,
  'card-width-small-x': 60,
  'card-width-small': 80,
  'card-width-medium': 136,
  'card-width-large': 247,

  'card-height-default': 100,
  'card-height-small-x': 72,
  'card-height-small': 80,
  'card-height-medium': 120,
  'card-height-large': 184,
  'card-border-color': 'rgba(240, 240, 245, 1)',

  'card-status-height': 184,
  'card-store-height': 256,

  'card-banner-width': 247,
  'card-banner-height': 120,

  'button-width-small': 88,
  'button-width': 160,
  'button-height': 48,
  'button-padding-vertical': 8,
  'button-padding-horizontal': 16,

  'radius-small': 6,
  'radius-medium': 12,
  'radius-large': 500,
};

const fonts = {
  Regular: 'IBMPlexSans_400Regular',
  Medium: 'IBMPlexSans_500Medium',
  SemiBold: 'IBMPlexSans_600SemiBold',
  Bold: 'IBMPlexSans_700Bold',
};

const fontScale = {
  tagline: 10,
  caption: 11,
  xxxs: 12,
  xxs: 14,
  xs: 16,
  sm: 18,
  md: 20,
  lg: 24,
  xl: 28,
  xxl: 32,
  xxxl: 36,
  display: 40,
};

export const theme = {
  card: {
    paddingDefault: gridScale.md,
    paddingSmall: gridScale.xxs,
    paddingLarge: gridScale.xxl,

    borderRadius: componentScale['radius-medium'],
    borderRadiusSmall: componentScale['radius-small'],
    borderRadiusLarge: componentScale['radius-large'],

    backgroundDefault: uiColors['white-default'],
    backgroundDanger: uiColors['danger-default'],
    backgroundDangerLighter: uiColors['danger-lighter'],

    backgroundSuccess: uiColors['success-default'],
    backgroundSuccessLighter: uiColors['success-lighter'],

    backgroundLighter: neutralColors.light['light-light'],
    backgroundDart: neutralColors.dark['dark-dark'],
  },
  screen: {
    background: uiColors['white-default'],

    paddingSmall: gridScale.xs,
    paddingDefault: gridScale.md,
    paddingLarge: gridScale.lg,
  },
  statusBar: {
    backgroundDefault: uiColors['white-default'],
    backgroundBack: uiColors['back-default'],
  },

  uiColors,
  neutralColors,
  fontScale,
  fonts,
};
