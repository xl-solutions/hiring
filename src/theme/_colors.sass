$shadow: #00000033
$header-bgc: #fff
$body-bgc: #f2f3ef
$darkGray: #414141
$blue: #00AECA
