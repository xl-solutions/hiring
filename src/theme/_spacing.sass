$spacing: (s: 8px, sm: 12px, m: 16px, l: 24px, lm: 32px, xl: 40px)