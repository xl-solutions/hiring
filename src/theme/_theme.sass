@import "./_colors"
@import "./_spacing"