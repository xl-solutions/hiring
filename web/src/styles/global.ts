import { createGlobalStyle } from 'styled-components';
import 'react-toastify/dist/ReactToastify.css';

export default createGlobalStyle`
  :root {
      --white: #FFFFFF;
      --green: #68F4A9;
      --black: #000000;
      --blue: #008FFB;
      --blue-dark: #0B3A5E;
      --gray-light: #f2f2f2;
      --gray-dark: #252731;
      --blue-light: #E7EFFF;
      --orange: #ff5f52;
      --red-dark: #AB4245;
      --gray: #252728;
      --gray-medium: #161819;
  }
  
  * {
    margin: 0;
    padding: 0;
    outline: 0;
  }

  body {
    background-color: #191920; 
    -webkit-font-smoothing: antialiased;
  }

  body, input, button {
    font: 14px Roboto, sans-serif;
    color: #FFFFFF;
  }

  #root {
    max-width: 1480px;
    margin: 0 auto;
    padding: 0 20px 50px;
  }

  button {
    cursor: pointer;
  }

  [disabled] {
    opacity: 0.6;
    cursor: not-allowed;
  }  
`;
