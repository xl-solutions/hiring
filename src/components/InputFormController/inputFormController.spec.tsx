import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '../../global/theme';

import { InputFormController } from './index';
import { RFValue } from 'react-native-responsive-fontsize';

const Providers: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

describe('Check component integrity InputFormController', () => {
  it('Should able to visible correctly perfection', () => {
    const { getByTestId } = render(
      <InputFormController
        testID="inputFormController-visible-perfection"
        iconName="search"
        isFocused={false}
        error={undefined}
        placeholder="Pesquisar ações, ETFs e outros para comprar"
        name="search"
      />,
      {
        wrapper: Providers,
      },
    );

    const inputFormControllerComponent = getByTestId(
      'inputFormController-visible-perfection',
    );

    expect(inputFormControllerComponent.props.style[0].backgroundColor).toEqual(
      'transparent',
    );

    expect(inputFormControllerComponent.props.placeholder).toEqual(
      'Pesquisar ações, ETFs e outros para comprar',
    );

    expect(inputFormControllerComponent.props.selectionColor).toEqual(
      'rgba(114, 114, 133, 1)',
    );
  });

  it('Should able to visible border to error', () => {
    const { getByTestId } = render(
      <InputFormController
        testID="inputFormController-error"
        iconName="search"
        isFocused={false}
        error="error-any"
        placeholder="Pesquisar ações, ETFs e outros para comprar"
        name="search"
      />,
      {
        wrapper: Providers,
      },
    );

    const inputFormControllerComponent = getByTestId(
      'inputFormController-error',
    );

    const fieldContainerComponent = getByTestId('test-fieldContainer');

    expect(inputFormControllerComponent.props.name).toEqual('search');

    expect(fieldContainerComponent.props.error).toEqual(true);

    expect(fieldContainerComponent.props.style[0].borderColor).toEqual(
      theme.uiColors['danger-default'],
    );

    expect(fieldContainerComponent.props.style[0].borderWidth).toEqual(
      RFValue(2),
    );
  });

  it('Should able to visible border to isFocused', () => {
    const { getByTestId } = render(
      <InputFormController
        testID="inputFormController-isFocused"
        iconName="search"
        isFocused={true}
        error={undefined}
        placeholder="Pesquisar ações, ETFs e outros para comprar"
        name="search"
      />,
      {
        wrapper: Providers,
      },
    );

    const inputFormControllerComponent = getByTestId(
      'inputFormController-isFocused',
    );

    const fieldContainerComponent = getByTestId('test-fieldContainer');

    expect(inputFormControllerComponent.props.name).toEqual('search');

    expect(fieldContainerComponent.props.error).toEqual(false);
    expect(fieldContainerComponent.props.isFocused).toEqual(true);

    expect(fieldContainerComponent.props.style[0].borderColor).toEqual(
      theme.uiColors['info-default'],
    );

    expect(fieldContainerComponent.props.style[0].backgroundColor).toEqual(
      theme.uiColors['info-lighter'],
    );

    expect(fieldContainerComponent.props.style[0].borderWidth).toEqual(
      RFValue(2),
    );
  });

  it('should able to visible Text Error', () => {
    const { getByTestId, toJSON } = render(
      <InputFormController
        testID="inputFormController-error-text"
        iconName="search"
        isFocused={false}
        error="error-any"
        placeholder="Pesquisar ações, ETFs e outros para comprar"
        name="search"
      />,
      {
        wrapper: Providers,
      },
    );

    const inputFormControllerComponent = getByTestId(
      'inputFormController-error-text',
    );

    const fieldContainerComponent = getByTestId('test-fieldContainer');
    const textErrorComponent = getByTestId('test-error');

    expect(inputFormControllerComponent.props).toHaveProperty('name');

    expect(textErrorComponent.props.children).toBe('error-any');

    expect(fieldContainerComponent.props.style[0].borderColor).toEqual(
      theme.uiColors['danger-default'],
    );

    expect(fieldContainerComponent.props.style[0].borderWidth).toEqual(
      RFValue(2),
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
