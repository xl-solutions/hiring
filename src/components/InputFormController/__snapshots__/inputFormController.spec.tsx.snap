// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check component integrity InputFormController should able to visible Text Error 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "paddingBottom": 31,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    error={true}
    isFocused={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F0F0F5",
          "borderColor": "#F53636",
          "borderRadius": 500,
          "borderStyle": "solid",
          "borderWidth": 4,
          "flexDirection": "row",
          "height": 98,
          "paddingBottom": 0,
          "paddingLeft": 53,
          "paddingRight": 53,
          "paddingTop": 0,
          "width": "100%",
        },
      ]
    }
    testID="test-fieldContainer"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#565666",
              "fontSize": 35,
            },
            Object {
              "marginRight": 29,
            },
          ],
          Object {
            "fontFamily": "feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      name="search"
      placeholder="Pesquisar ações, ETFs e outros para comprar"
      rejectResponderTermination={true}
      selectionColor="rgba(114, 114, 133, 1)"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "color": "#565666",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "IBMPlexSans_500Medium",
            "fontSize": 31,
            "fontStyle": "normal",
            "fontWeight": "500",
            "height": 90,
            "lineHeight": 40,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 28,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 28,
            "textAlign": "left",
          },
        ]
      }
      testID="inputFormController-error-text"
      underlineColorAndroid="transparent"
    />
    <Text
      allowFontScaling={false}
      error="error-any"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#F53636",
              "fontSize": 35,
            },
            Object {
              "marginLeft": 29,
            },
          ],
          Object {
            "fontFamily": "feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#F53636",
          "fontFamily": "IBMPlexSans_500Medium",
          "fontSize": 24,
          "fontStyle": "normal",
          "fontWeight": "500",
          "lineHeight": 28,
          "paddingLeft": 47,
          "paddingTop": 8,
          "textAlign": "left",
        },
      ]
    }
    testID="test-error"
  >
    error-any
  </Text>
</View>
`;
